Initial Commit

vrushang1234 opened the pull request:
### **Initial Commit for NewGUI**
**Closes #2 **
- Converted the Javascript code into typescript
- All of the values shown on the GUI currently are static values and haven't used any states to change them in real-time (hardcoded)

taesungh commented on control-station/src/components/SensorBoxes/Sensors/style.css:
```diff
@@ -0,0 +1,25 @@
+#sensorbox{
+    height: 25vh;
+    width: 40%;
```
Suggestion [repeating]: using viewport proportions or percentages assumes this content will be displayed in a highly particular layout with specific content dimensions. Consider using a flex or grid-based approached.

taesungh replied:
Claiming okay for now but will want to revisit this.

taesungh commented on control-station/src/components/SensorBoxes/SensorBoxContainer.tsx:
```diff
@@ -0,0 +1,11 @@
+import CCcontainer from "./Camera&Console/CCcontainer"
+import SensorContainer from "./Sensors/SensorsContainer"
+import "./style.css"
+export default function SensorBoxContainer(){
```
Suggestion: the name `SensorBoxContainer` is a bit verbose, consider `Sensors` or `SensorData`

taesungh commented on control-station/src/components/SensorBoxes/Camera&Console/Camera.tsx:
```diff
@@ -0,0 +1,8 @@
+import "./style.css"
+export default function Camera(){
+    return(
+        <div id="camera">
+
+        </div>
+    )
+}
```
Thought [repeating]: unsure if the Camera and Console would categorically fit under "SensorBoxes". Consider extracting to separate components.

taesungh commented on control-station/src/components/ControlPanel/style.css:
```diff
@@ -0,0 +1,30 @@
+    bottom: 0;
+    width: 100%;
+    background-color: black;
+    height: 9%;
```
Suggestion [repeating, also for Navbar]: using a percentage for height makes the element too short on narrower views. Consider relying on the content and padding to determine the height instead of explicitly setting it.

taesungh commented on control-station/src/components/ControlPanel/style.css:
```diff
@@ -0,0 +1,30 @@
+}
+
+.button{
+    height: 60px;
+    width: 200px;
```
Issue: Using an explicit `height` and `width` results in content spilling over on shorter views and is not entirely flexible depending on the inner content. Consider using `padding` instead of explicit dimensions.

taesungh commented on control-station/src/components/Navbar/Navbar.tsx:
```diff
@@ -0,0 +1,9 @@
+import "./style.css"
+import HX from "../../../public/Data/Images/HX Logo.png"
```
Chore: assets in `public` cannot be imported from JavaScript ([Static Asset Handling, The `public` directory](https://vitejs.dev/guide/assets.html#the-public-directory)). Consider moving this file to `src/assets/` to take advantage of webpack's file loader.

samderanova replied:
We might also want to consider using [path aliases](https://vitejs.dev/config/shared-options#resolve-alias) so we don't need to specify full relative paths.

taesungh commented on control-station/src/components/Navbar/Navbar.tsx:
```diff
@@ -0,0 +1,9 @@
+export default function Navbar(){
+    return(
+        <header id="navbar">
+            <img src={HX} style={{height:"60px"}}/>
```
Nitpick: image is missing alternative text.

taesungh reviewed (changes requested):
Thanks for working on this. Several notes about using class names instead of ids, more flexible CSS styles, and reorganizing files. Additionally, please reformat source files with Prettier.

samderanova commented on control-station/src/components/Navbar/style.css:
```diff
@@ -0,0 +1,11 @@
+    height: 10%;
+    font-size: 2rem;
+    background-color: black;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    color: white;
+    font-weight: 700;
```
@taesungh, should we add Bootstrap here or save that for another issue?

vrushang1234 commented:
Fixed most of the issues except using classes instead of IDs as I tried to reuse the classes as much as possible in the initial commit. Also added a state and context for console list.

taesungh commented on control-station/src/components/controlpanel/controlpanel.css:
Chore: for consistency, stylesheet filename should match component as CamelCase.

taesungh reviewed (changes requested):
Please try to limit the scope of changes: we were looking to review the preliminary visual design and file structure before incorporating functionality. As such, could you revert the changes on adding the context and state?

There are still some lingering comments from the first review, including reformatting files with Prettier.

vrushang1234 commented:
> Please try to limit the scope of changes: we were looking to review the preliminary visual design and file structure before incorporating functionality. As such, could you revert the changes on adding the context and state?
> 
> 
> 
> There are still some lingering comments from the first review, including reformatting files with Prettier.

I use prettier locally. Do you mean I have to add Prettier to the eslint?

taesungh commented:
Perhaps something went wrong locally: the TSX and CSS files do not appear to be formatted. You can also use the [Prettier extension for VS Code](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode).

samderanova commented:
Additionally, we might want to consider having some sort of [`.prettierrc`](https://prettier.io/docs/en/configuration.html) configuration so that the formatting is consistent for all of us.

taesungh commented:
> Additionally, we might want to consider having some sort of [`.prettierrc`](https://prettier.io/docs/en/configuration.html) configuration so that the formatting is consistent for all of us.

Included in `control-station/.prettierrc` from d5cc776.

samderanova commented:
Sorry I thought it wasn't included since I didn't see it in the list of changes for this PR. Please disregard my comment.

taesungh commented on control-station/vite.config.ts:
```diff
@@ -7,6 +7,7 @@ export default defineConfig({
 	resolve: {
 		alias: {
 			"@/": "/src/",
+			"$/":"/public/"
```
Chore [repeating]: unused path alias.

taesungh commented on control-station/src/views/Dashboard/Dashboard.tsx:
```diff
@@ -1,4 +1,4 @@
-import { Status } from "@/components";
+import { Status } from "@/Components";
```
Chore: please keep the actual `components` directory name lowercase.

taesungh commented on control-station/src/App.tsx:
```diff
@@ -1,13 +1,27 @@
+}
 
-import "./App.css";
+export const consoleContext = createContext<ConsoleContextProps | undefined>(
+	undefined
+);
```
As mentioned before, let's focus on the preliminary visual design and limit the scope otherwise. Please remove the context for now.

SM-dot commented on control-station/src/components/Navbar/Navbar.tsx:
While inline styling (style={{ height: "60px" }}) is acceptable for a small adjustment, consider moving such styling to the external stylesheet for better separation of concerns.

taesungh commented on control-station/vite.config.ts:
```diff
@@ -6,7 +6,7 @@ export default defineConfig({
 	resolve: {
 		alias: {
-			"@/": "/src/",
+			"@/": "/src/"
```
Chore: let's keep the trailing comma for future diffs

taesungh commented on control-station/index.html:
```diff
@@ -4,6 +4,9 @@
 		<meta name="viewport" content="width=device-width, initial-scale=1.0" />
+		<link rel="preconnect" href="https://fonts.googleapis.com">
+		<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+		<link href="https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,100;0,300;0,400;0,500;0,700;0,900;1,100;1,300;1,400;1,500;1,700;1,900&display=swap" rel="stylesheet">
```
Thought: including all of the font weights incurs a bit of unnecessary bandwidth. I think we could remove the extremal ones (100 and 900) and keep the rest from 300 to 700.

taesungh reviewed (approved):
We'll want to revisit some of the styling concerns (perhaps look into Bootstrap as Sam suggested) and component hierarchy (particularly within `SensorBoxes`), but this will work for now.

taesungh commented on control-station/src/components/index.ts:
```diff
@@ -1 +1,4 @@
+export { default as ControlPanel } from "./ControlPanel/ControlPanel";
+export { default as Navbar } from "./Navbar/Navbar";
+export { default as SensorData } from "./SensorBoxes/SensorData";
+export { default as Status } from "./status/Status";
```
Issue: component directory name for `Status` needs to be capitalized
```suggestion
export { default as Status } from "./Status/Status";
```
